src/MongoDB.Bson/ObjectModel/ObjectId.cs

```diff
@@ -419,19 +429,19 @@ private static short GetPid()
             }
         }
 
-        private static int GetTimestampFromDateTime(DateTime timestamp)
+        private static uint GetTimestampFromDateTime(DateTime timestamp)
         {
             var secondsSinceEpoch = (long)Math.Floor((BsonUtils.ToUniversalTime(timestamp) - BsonConstants.UnixEpoch).TotalSeconds);
-            if (secondsSinceEpoch < int.MinValue || secondsSinceEpoch > int.MaxValue)
+            if (secondsSinceEpoch < uint.MinValue || secondsSinceEpoch > uint.MaxValue)
             {
                 throw new ArgumentOutOfRangeException("timestamp");
             }
-            return (int)secondsSinceEpoch;
+            return (uint)secondsSinceEpoch;
```

Comment:
Instead of changing the return type from `int` to `uint` we can just change the calculation to get the result we need:

    private static int GetTimestampFromDateTime(DateTime timestamp)
    {
        var secondsSinceEpoch = (long)Math.Floor((BsonUtils.ToUniversalTime(timestamp) - BsonConstants.UnixEpoch).TotalSeconds);
        if (secondsSinceEpoch < uint.MinValue || secondsSinceEpoch > uint.MaxValue)
        {
            throw new ArgumentOutOfRangeException("timestamp");
        }
        return (int)(uint)secondsSinceEpoch;
    }

Reply:
Also... I can't comment in the right place because github only lets me comment on changes, but the `CreationTime` property needs to be changed to:

    public DateTime CreationTime
    {
        get { return BsonConstants.UnixEpoch.AddSeconds((uint)Timestamp); }
    }